rest/route/keys_routes.go

```diff
@@ -146,3 +147,49 @@ func (h *keysPostHandler) Execute(ctx context.Context, sc data.Connector) (Respo
 
 	return ResponseData{}, nil
 }
+
+type keysDeleteHandler struct {
+	keyName string
+}
+
+func getKeysDeleteRouteManager(route string, version int) *RouteManager {
+	return &RouteManager{
+		Route: route,
+		Methods: []MethodHandler{
+			MethodHandler{
+				PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+				Authenticator:     &RequireUserAuthenticator{},
+				RequestHandler:    &keysDeleteHandler{},
+				MethodType:        http.MethodDelete,
+			},
+		},
+		Version: version,
+	}
+}
+
+func (h *keysDeleteHandler) Handler() RequestHandler {
+	return &keysDeleteHandler{}
+}
+
+func (h *keysDeleteHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	vars := mux.Vars(r)
+	h.keyName = vars["key_name"]
+	if strings.TrimSpace(h.keyName) == "" {
+		return &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "empty key name",
+		}
+	}
+
+	return nil
+}
+
+func (h *keysDeleteHandler) Execute(ctx context.Context, sc data.Connector) (ResponseData, error) {
+	user := MustHaveUser(ctx)
+
+	if err := sc.DeletePublicKey(user, h.keyName); err != nil {
+		return ResponseData{}, errors.New("couldn't delete key")
+	}
+
+	return ResponseData{}, nil
```

Comment:
Can we return something more useful here, like the current list of keys for the user?

Reply:
I understand why that might be useful, but I really don't like the idea of mixing DELETE and GET responsibilities into one route. Status code 200 OK is documented.